src/osd.c

```diff
@@ -223,6 +223,138 @@ preview_cycled_view(struct view *view)
 	wlr_scene_node_raise_to_top(osd_state->preview_node);
 }
 
+static void
+osd_draw(cairo_t *cairo, int w, int h, struct wl_list *node_list,
+	int current_view_index, struct theme *theme, bool show_workspace,
+	const char *workspace_name)
+{
+	struct wlr_scene_node *node;
+	cairo_surface_t *surf = cairo_get_target(cairo);
+
+	/* Draw background */
+	set_cairo_color(cairo, theme->osd_bg_color);
+	cairo_rectangle(cairo, 0, 0, w, h);
+	cairo_fill(cairo);
+
+	/* Draw border */
+	set_cairo_color(cairo, theme->osd_border_color);
+	draw_cairo_border(cairo, w, h, theme->osd_border_width);
+
+	/* Highlight current window */
+	if (current_view_index != -1) {
```

Comment:
This can be removed completely and replaced by a simpler check further down below. See next comment.